Entry for Java String `.valueOf()` (#1171)
Codecademy/docs: content/java/concepts/strings/terms/valueOf/valueOf.md

---
Title: '.valueOf()'

> Description: 'Returns the string representation of given value.'

Reviewer: ```suggestion
'Returns the string representation of a given value.'
```

Author: Hey @Dusch4593, Do I have to remove the Description tag too?

Subjects:
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`String.valueOf()`** is a built-in java function for converting different types of values into its corresponding `string` representation.

Reviewer: ```suggestion
The **`String.valueOf()`** method returns the string representation of a given value, including ones of a different [data type](https://www.codecademy.com/resources/docs/java/data-types).
```

## Syntax

```pseudo

> String.valueOf(int i)         // Returns the string representation of int i.
> String.valueOf(long l)        // Returns the string representation of long l.
> String.valueOf(boolean b)     // Returns the string representation of boolean argument.
> String.valueOf(float f)       // Returns the string representation of float f.
> String.valueOf(double df)     // Returns the string representation of double argument.

Reviewer: ```suggestion
String.valueOf(value)
```

For the "Syntax" section, I would stick with 1-2 lines for the \```pseudo code block (with little to no comments) and then add a sentence underneath the goes something like:

> The \`value\` parameter can be of type \`int\`, \`long\`, \`boolean\`, \`float\`, and \`double\`.

```

## Example

> This following example returns the string representation with the `.valueOf()` method:

Reviewer: ```suggestion
The following is an example of the `.valueOf()` method being applied to values of different data types:
```

```java

> // Main.java
> public class Main {
>   public static void main(String[] args) {
>       // Concatenating an integer with string
>       int x = 20;
>       String str = "22";
>       System.out.println(String.valueOf(x) + str);
> 
>       // String representation of boolean argument
>       boolean b = true;
>       System.out.println(String.valueOf(b));
> 
>       // Concatenating string with float
>       float f = 4.66f;
>       System.out.println(str + String.valueOf(f));
> 
>       // Char to String
>       char ch = 'A';
>       System.out.println(String.valueOf(ch));
>   }
> }

Reviewer: ```suggestion
// Main.java
public class Main {
  public static void main(String[] args) {
    // Concatenating an integer with string
    int x = 20;
    String str = "22";
    System.out.println(String.valueOf(x) + str);
    
    // String representation of boolean argument
    boolean b = true;
    System.out.println(String.valueOf(b));
    
    // Concatenating string with float
    float f = 4.66f;
    System.out.println(str + String.valueOf(f));
    
    // Char to String
    char ch = 'A';
    System.out.println(String.valueOf(ch));
  }
}
```
